From 6a6479236983df37753367da96d64d17296286b5 Mon Sep 17 00:00:00 2001
From: Hatato <jesse12p@gmail.com>
Date: Wed, 28 Oct 2020 20:27:16 +0000
Subject: [PATCH] Fix generic URL regex.

The domain part matched / and other extra characters, which breaks
generic URL dispatching.

Signed-off-by: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
---
 service/dispatcher.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/service/dispatcher.c b/service/dispatcher.c
index a14c76a..3124e37 100644
--- a/service/dispatcher.c
+++ b/service/dispatcher.c
@@ -570,7 +570,7 @@ dispatcher_init (GMainLoop * mainloop, OverlayTracker * intracker)
 
 	applicationre = g_regex_new("^application:///([a-zA-Z0-9_\\.-]*)\\.desktop$", 0, 0, NULL);
 	appidre = g_regex_new("^appid://([a-z0-9\\.-]*)/([a-zA-Z0-9-\\.]*)/([a-zA-Z0-9\\.-]*)$", 0, 0, NULL);
-	genericre = g_regex_new("^([a-z][a-z0-9]*):(?://(?:.*@)?([a-zA-Z0-9\\.-_]*)(?::[0-9]*)?/?)?(.*)?$", 0, 0, NULL);
+	genericre = g_regex_new("^([a-z][a-z0-9]*):(?://(?:.*@)?([a-zA-Z0-9\\._-]*)(?::[0-9]*)?/?)?(.*)?$", 0, 0, NULL);
 	intentre = g_regex_new("^intent://.*package=([a-zA-Z0-9\\.]*);.*$", 0, 0, NULL);
 
 	g_bus_get(G_BUS_TYPE_SESSION, cancellable, bus_got, mainloop);
-- 
2.30.2

